src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,23 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+export const booksSlice = createSlice({
+  name: 'books',
+  initialState: {
+    books: [
+    ],
+  },
+  reducers: {
+    add: (state, action) => {
+      state.books.push(action.payload);
+    },
+    remove: (state, action) => {
+      if (action.payload > -1) {
+        state.books.splice(action.payload, 1);
+      }
```

Comment:
- [ ] Please note that the `books` state is meant to hold an array of book objects and the `action.payload` will receive the ID of the book to be removed so when you do `state.books.splice(action.payload, 1);` you're searching for and trying to remove an ID from the books array but as I mentioned the array doesn't hold IDs but instead objects so It wouldn't work as you expected it.
- [ ] As mentioned previously it would be better If you can try the below code or a similar implementation
```jsx
state.books = state.books.filter((book) => book.id !== action.payload);
```